fe/fe-core/src/main/java/org/apache/doris/nereids/memo/GroupExpression.java

```diff
@@ -168,6 +169,11 @@ public boolean equals(Object o) {
             return false;
         }
         GroupExpression that = (GroupExpression) o;
+        // if the plan is LogicalOlapScan, this == that should be true,
+        // because equals() can not divide UnboundRelation and the plan above.
```

Comment:
this comment is not clear to describe the question. we need to explain that if one relation appear in plan more than once, we cannot distinguish them throw equals function, since equals function cannot use output info.